Python/pywarpx/fields.py

```diff
@@ -279,7 +279,7 @@ def __getitem__(self, index):
             raise Exception('Too many indices given')
 
         # Expand the indices to length 3
-        ii = self._get_indices(index, None)
+        ii = self._get_indices(index, slice(None))
```

Comment:
@dpgrote I've been trying to get the test in `Examples/Physics_applications/capacitive_discharge/PICMI_inputs_1d.py` to run, but I encountered an error in the call to `fields.py` to grab `rho` due to this line. The error was:
```
  File "/home/roelof/scratch/build/warpx/warpx_pybind/lib/site-packages/pywarpx/fields.py", line 179, in _find_start_stop
    iistop = ii + 1
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```
which is circumvented by this change but you might prefer a different fix (perhaps dealing with the None in `_find_start_stop()`.

With this fix the test unfortunately still doesn't run due to errors in the part that writes to `phi`.

Reply:
This fix is good, thanks! I also just did a push to make the same fix for the `__setitem__`. Can you check if this fixes the problem when writing `phi`?

Reply:
Sorry, I should've been more clear. With that same fix in the `__setitem__` function there is still an error in the part that writes to `phi`:
```
  File "PICMI_inputs_1d.py", line 105, in _run_solve
    fields.PhiFPWrapper(0, True)[...] = self.phi
  File "/home/roelof/scratch/build/warpx/warpx_pybind/lib/site-packages/pywarpx/fields.py", line 390, in __setitem__
    mf_arr[block_slices] = value3d[global_slices]
IndexError: too many indices for array: array is 1-dimensional, but 3 were indexed
```

Reply:
In this specific case `mf_arr` has shape (131, 1, 1, 1) while in the current execution `value3d` (i.e. `phi`) has shape (131). The test runs if I change in the input script
```
fields.PhiFPWrapper(0, True)[...] = self.phi[:]
```
to
```
fields.PhiFPWrapper(0, True)[...] = self.phi[:,None,None,None]
```
But is this what we would want?

Reply:
Also, just to let you know, I'm seeing that `__getitem__` gives an array of the correct length but always with all the values in the array the same (at least in the 1d case I'm working on).

Reply:
I just pushed some changes that should fix this issues with the get and set item methods. Also, for convenience, I added the `shape` attribute, similar to the  numpy shape, but here it returns the shape of the global array.

Reply:
Thanks! Looks like the 1d test case is happy now :slightly_smiling_face: